addons/sale/models/sale_order.py

```diff
@@ -1543,13 +1545,13 @@ def _create_invoices(self, grouped=False, final=False, date=None):
         if not grouped:
             new_invoice_vals_list = []
             invoice_grouping_keys = self._get_invoice_grouping_keys()
-            invoice_vals_list = sorted(
+            invoice_vals_list_grouped = dict(groupby(
                 invoice_vals_list,
-                key=lambda x: [
-                    x.get(grouping_key) for grouping_key in invoice_grouping_keys
-                ]
-            )
-            for _grouping_keys, invoices in groupby(invoice_vals_list, key=lambda x: [x.get(grouping_key) for grouping_key in invoice_grouping_keys]):
+                key=lambda x: tuple(x.get(grouping_key) for grouping_key in invoice_grouping_keys),
+            ))
+
+            for grouping_key in sorted(invoice_vals_list_grouped):
+                invoices = invoice_vals_list_grouped[grouping_key]
```

Comment:
I'm not sure to understand this change and how it's related to your PR
Did the groupby lose the sorting order? :thinking:

Reply:
Thanks for having a look :smile: 
This diff isn't necessary (because I imported `odoo.tools`'s `groupby`, it then clashed with `itertools`'s groupby), but I've fixed that now and removed this diff.

The PR isn't quite ready yet, but we are aiming to merge it this week if possible.
I was also thinking of possibly factorizing the `_extend_with_attachments` method in a mixin inherited by both `account.move` and `sale.order` (see my last commit), what would you think of that?